coalib/processes/Processing.py

```diff
@@ -267,16 +267,17 @@ def get_file_dict(filename_list, log_printer=None, allow_raw_files=False):
     :return:                Reads the content of each file into a dictionary
                             with filenames as keys.
     """
-    file_dict = {}
+    file_dict = FileDict()
     for filename in filename_list:
         try:
-            with open(filename, 'r',
-                      encoding=detect_encoding(filename)) as _file:
```

Comment:
I see that FileFactory is not using `detect_encoding`, and is always using `'utf-8'`.  I suspect that is why one of your tests is failing.

Anyway, that is a bug and needs to get fixed.

Reply:
Makman had raised this issue in an [earlier review](https://github.com/coala/coala/pull/5627#discussion_r203849101) and said that we can do it later but I will try to use `detect_encoding` in `FileFactory` locally and see if the tests pass.

Reply:
`FileFactory` aims to be consistent and simple, that was my motivation to fix the encoding. I have thought about different encodings and using `detect_encoding` but wanted to make it possible at the same time to control encodings as well (would e.g. allow to write encoding converter bears). Haven't yet decided finally^^

If it helps for now to make tests pass it's fine with me, but always using `detect_encoding` is a workaround for me^^

Reply:
ok, lets create issues about this and work on them after gsoc.  @palash25 you can mention this in your work report under 'things to be done'